docs/Customization/Custom-Chisel.rst

```diff
@@ -0,0 +1,38 @@
+.. _custom_chisel:
+
+Integrating into the Generator Build System
+===========================================
+
+While developing, you want to include Chisel code in a submodule so that it can be shared by different projects.
+To add a submodule to the Chipyard framework, make sure that your project is organized as follows.
+
+.. code-block:: none
+
+    yourproject/
+        build.sbt
+        src/main/scala/
+            YourFile.scala
+
+Put this in a git repository and make it accessible.
```

Comment:
I don't think you want to force people to make it a git repository. Instead, I would move the part about putting it in a git repo towards the bottom and make it known that it is optional but recommended.

Reply:
I think if we reorder the submodule part to the bottom, people following step-by-step will first create their custom files in a normal directory. Creating a git submodule out of a normal directory that already exists is nontrivial.

Instead I will add stronger language here explaining that while the submodule approach is optional, it is heavily recommended.

Reply:
If you're going into this level of detail, you might as well describe what needs to be in your custom build.sbt....

Reply:
(like, an example of a minimal custom build.sbt)